client/lib/format-number-compact/index.js

```diff
@@ -0,0 +1,46 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import { numberFormat } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { THOUSANDS } from './thousands';
+
+/**
+ * Formats a number to a short format given a language code
+ * @param   {Number}     number              number to format
+ * @param   {String}     code                language code e.g. 'es'
+ * @returns {?String}                        A formatted string.
+ */
+export default function formatNumberCompact( number, code ) {
+	//use numberFormat directly from i18n in this case!
+	if ( isNaN( number ) || ! THOUSANDS[ code ] ) {
+		return null;
```

Comment:
Not exactly sure how this would work.
Since we have `numberFormat` available, why not directly returning `numberFormat( number )` (which is what I'd expect from the inline comment above, btw)?